Bottom: 1cb67c9e7c676d1b52add5ff13d7e723e5c28b1c
Top:    a600890d1ce33a8d974029b612b737256e44ccc7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:57:50 +0000

polygon.c: Refactor NoHolesPolygonDicer to call poly_Copy0 rather than poly_Copy1

poly_Copy1 is much more of an internal function to polygon1.c, and this is its
only external caller. Replace this occurance with poly_Copy0, which is intended
to be the public API.


---

diff --git a/src/polygon.c b/src/polygon.c
index 57cfe80..a892dce 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -2011,9 +2011,8 @@ NoHolesPolygonDicer (PolygonType *p, const BoxType * clip,
 {
   POLYAREA *main_contour, *cur, *next;
 
-  main_contour = poly_Create ();
   /* copy the main poly only */
-  poly_Copy1 (main_contour, p->Clipped);
+  poly_Copy0 (&main_contour, p->Clipped);
   /* clip to the bounding box */
   if (clip)
     {
